drivers/input/input_gpio_kbd_matrix.c

```diff
@@ -205,7 +243,16 @@ static int gpio_kbd_matrix_init(const struct device *dev)
 	LOG_DBG("direct_read: %d direct_write: %d",
 		data->direct_read, data->direct_write);
 
-	return input_kbd_matrix_common_init(dev);
+	ret = input_kbd_matrix_common_init(dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	if (cfg->gpio_cb == NULL && cfg->idle_poll_dwork == NULL) {
```

Comment:
Nit - add a comment that scanning mode selected on this path.  It's non-obvious that `gpio_cb` implies interrupt mode and that `idle_poll_dwork` implies polling mode.